nexus/db-queries/src/db/queries/virtual_provisioning_collection_update.rs

```diff
@@ -399,129 +108,359 @@ impl VirtualProvisioningCollectionUpdate {
     // - Project
     // - Silo
     // - Fleet
-    //
-    // Arguments:
-    // - do_update: A boolean SQL query to answer the question: "Should this update
-    // be applied"? This query is necessary for idempotency.
-    // - update: A SQL query to actually modify the resource record. Generally
-    // this is an "INSERT", "UPDATE", or "DELETE".
-    // - project_id: The project to which the resource belongs.
-    // - values: The updated values to propagate through collections (iff
-    // "do_update" evaluates to "true").
-    fn apply_update<U, V>(
+    fn apply_update(
         update_kind: UpdateKind,
-        update: U,
         project_id: uuid::Uuid,
-        values: V,
-    ) -> Self
-    where
-        U: QueryFragment<Pg> + crate::db::subquery::Subquery + Send + 'static,
-        V: diesel::AsChangeset<Target = virtual_provisioning_collection::table>,
-        <V as diesel::AsChangeset>::Changeset:
-            QueryFragment<Pg> + Send + 'static,
-    {
-        let parent_silo = ParentSilo::new(project_id);
-        let all_collections = AllCollections::new(
-            project_id,
-            &parent_silo,
-            *crate::db::fixed_data::FLEET_ID,
-        );
-
-        let quotas = Quotas::new(&parent_silo);
-        let silo_provisioned = SiloProvisioned::new(&parent_silo);
+    ) -> TypedSqlQuery<SelectableSql<VirtualProvisioningCollection>> {
+        let query = QueryBuilder::new().sql("
+WITH
+  parent_silo AS (SELECT project.silo_id AS id FROM project WHERE project.id = ").param().sql("),")
+            .bind::<sql_types::Uuid, _>(project_id).sql("
```

Comment:
IIUC it shouldn't really be necessary to format the query strings here identically to the `format_sql` output, since we're still going to call `format_sql` on them prior to testing that they match?